python/lwt_interface/example_c_module.c

```diff
@@ -38,7 +38,53 @@
 
 #include "tskit_lwt_interface.h"
 
+static PyObject * example_receiving(PyObject *self, PyObject *args) {
+    PyObject* ret = NULL;
+    LightweightTableCollection *tables = NULL;
+
+    if (!PyArg_ParseTuple(args, "O!", &LightweightTableCollectionType, &tables)) {
+        goto out;
+    }
+    /* Do something simple with the tables for the purposes of example */
+    if (tables->tables->nodes.num_rows > 2) {
+        ret = Py_True;
+    } else {
+        ret = Py_False;
+    }
+
+out:
+    return ret;
+}
+
+static PyObject * example_returning(PyObject *self, PyObject *args) {
+    PyObject * ret = NULL;
+    int err = -1;
+    LightweightTableCollection *tables = (LightweightTableCollection *)PyObject_CallFunction(
```

Comment:
This is nice. I wonder if it's simpler to show an example where the C code modifies the tables? We don't strictly need to return a new TableCollection since we can do this easily create a new one in Python and pass it in.

But, if you're sure the refcounting is all solid here, then this is good too.

Reply:
I think it best if recommend always creating table collections in Python - I've added a note to the docs about this and changed to example to be a modifying example.